youtube_dl/extractor/lcp.py

```diff
@@ -0,0 +1,122 @@
+    IE_NAME = 'LCP'
+    _VALID_URL = r'https?://(?:www\.)?lcp\.fr/(?:[^\/]+/)*(?P<id>[^/]+)'
+
+    _TESTS = [{
+        'url': 'http://www.lcp.fr/la-politique-en-video/schwartzenberg-prg-preconise-francois-hollande-de-participer-une-primaire',
+        'md5': 'aecf5a330cfc1061445a9af5b2df392d',
+        'info_dict': {
+            'id': 'd56d03e9',
+            'url': 're:http://httpod.scdn.arkena.com/11970/d56d03e9_[0-9]+.mp4',
+            'ext': 'mp4',
+            'title': 'Schwartzenberg (PRG) préconise à François Hollande de participer à une primaire à gauche'
+        }
+    }, {
+        'url': 'http://www.lcp.fr/emissions/politique-matin/271085-politique-matin',
+        'md5': '6cea4f7d13810464ef8485a924fc3333',
+        'info_dict': {
+            'id': '327336',
+            'url': 're:http://httpod.scdn.arkena.com/11970/327336_[0-9]+.mp4',
+            'ext': 'mp4',
+            'title': 'Politique Matin - Politique matin'
+        }
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+
+        # Extract the required info of the media files gathered in a dictionary
+        media_files_info = self.__extract_from_webpage(display_id, webpage)
+        # Some web pages embed videos from other platforms like dailymotion, therefore we pass on these URLs
+        if media_files_info is None:
+            return self.url_result(url, 'Generic')
+
+        video_formats = self.__get_video_formats(media_files_info)
+        video_thumbnails = self.__get_thumbnails(media_files_info)
+
+        return {
+            'id': media_files_info['EntryName'],
+            'title': self._og_search_title(webpage),
+            'formats': video_formats,
+            'thumbnails': video_thumbnails
+        }
+
+    def __extract_from_webpage(self, display_id, webpage):
+        """Extracts the media info JSON object for the video for the provided web page."""
+        embed_url = self.__extract_embed_url(webpage)
+        embed_regex = r'(?:[a-zA-Z0-9]+\.)?lcp\.fr/embed/(?P<clip_id>[A-za-z0-9]+)/(?P<player_id>[A-za-z0-9]+)/(?P<skin_name>[^\/]+)'
+
+        clip_id = self._search_regex(embed_regex, embed_url, 'clip id', group='clip_id', fatal=False)
+        player_id = self._search_regex(embed_regex, embed_url, 'player id', group='player_id', fatal=False)
+        skin_name = self._search_regex(embed_regex, embed_url, 'skin name', group='skin_name', fatal=False)
+
+        # Check whether the matches failed, which might be when dealing with other players (e.g., dailymotion stream)
+        if (clip_id is None) or (player_id is None) or (skin_name is None):
+            return None
+
+        return self.__extract_from_player(display_id, clip_id, player_id, skin_name)
+
+    def __extract_embed_url(self, webpage):
+        return self._search_regex(
+            r'<iframe[^>]+src=(["\'])(?P<url>.+?)\1',
+            webpage, 'embed url', group='url')
```

Comment:
This will fail in case of no iframe. Generic embeds may be embedded differently so flow will never reach generic extraction delegation for such embeds.